src/firefly/java/edu/caltech/ipac/firefly/server/query/tables/IpacTableFromSource.java

```diff
@@ -99,41 +108,51 @@ public boolean doCache() {
     private File getSourceFile(String source, TableServerRequest request, boolean checkForUpdates) {
         if (source == null) return null;
         try {
-            URL url = makeUrl(source);
-            if (url == null) {
-                File f = ServerContext.convertToFile(source);
-                if (f == null || !f.canRead()) return  null;
-
-                StringKey key = new StringKey(getUniqueID(request), f.lastModified());
-                File cached  = (File) getCache().get(key);
-                if (cached != null) return cached;    // it's cached.. return it.
-
-                File res = convertToIpacTable(f, request);
-                getCache().put(key, res);
-                return res;
-            } else {
-                StringKey key = new StringKey(getUniqueID(request), url);
-                File res  = (File) getCache().get(key);
-
-                String ext = FileUtil.getExtension(url.getPath());
-                ext = StringUtils.isEmpty(ext) ? ".ul" : "." + ext;
-                File nFile = createFile(request, ext);
-
-                HttpURLConnection conn = (HttpURLConnection) URLDownload.makeConnection(url);
-                if (res == null) {
-                    URLDownload.getDataToFile(conn, nFile, null, false, true, false, Long.MAX_VALUE);
-                    res = convertToIpacTable(nFile, request);
+            if (isWorkspace(request)) {
+                WsServerParams wsParams = new WsServerParams();
+                wsParams.set(WsServerParams.WS_SERVER_PARAMS.CURRENTRELPATH, source);
+                WsServerUtils wsUtil= new WsServerUtils();
+                String s=  wsUtil.upload(wsParams);
+                return ServerContext.convertToFile(s);
```

Comment:
convertToIpacTable() should also be called here as well for the same reason as above.